smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/middlewares/providers/HttpQueryItemProvider.kt

```diff
@@ -73,9 +73,13 @@ class HttpQueryItemProvider(
         writer.openBlock("extension \$N: \$N {", "}", inputSymbol, ClientRuntimeTypes.Middleware.Providers.QueryItemProvider) {
             writer.openBlock("public var queryItems: [\$N] {", "}", ClientRuntimeTypes.Core.URLQueryItem) {
                 writer.openBlock("get throws {", "}") {
-                    writer.write("var items = [\$N]()", ClientRuntimeTypes.Core.URLQueryItem)
-                    generateQueryItems()
-                    writer.write("return items")
+                    if (queryLiterals.isNotEmpty() || queryBindings.isNotEmpty()) {
+                        writer.write("var items = [\$N]()", ClientRuntimeTypes.Core.URLQueryItem)
+                        generateQueryItems()
+                        writer.write("return items")
+                    } else {
+                        writer.write("return []")
+                    }
```

Comment:
there should be a test on this.

Reply:
Good call, because in writing the test, I discovered that there is potentially a bug with the query item provider logic.

I am reverting this change in this PR and will investigate further; this fix will be filed in a separate PR once I have diagnosed the issue.